Updates MC documentation towards updates (#837)
giantswarm/docs: src/content/general/management-clusters/index.md

Our workload clusters are versioned using [workload cluster releases]({{< relref "/general/releases" >}}). From a workload cluster point of view, upgrades are described in [cluster upgrades]({{< relref "/general/cluster-upgrades" >}}).

> When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io/" >}}) and any new operator versions are deployed to the management cluster.

Reviewer: ```suggestion
When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.
```

I think it just needs the file extension.

Author: lol, apparently so. thanks!

> When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io/" >}}) and any new operator versions are deployed to the management cluster.

Reviewer 2: ```suggestion
When we publish a new workload cluster release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io/" >}}) and any new operator versions are deployed to the management cluster.
```

Author: i don't know how, but this got resolved somewhere :D

> When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.

Reviewer 2: ```suggestion
When we publish a new workload cluster release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.
```

Author: i don't know how git works, commited 🚀

> When we publish a new workload cluster release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.

Reviewer 2: ```suggestion
When we publish a new workload cluster release, we create a new [Release]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) custom resource and any new operator versions are deployed to the management cluster.
```

No need to capitalize custom resource

We deploy new instances of the operators with this new version, via the [release-operator](https://github.com/giantswarm/release-operator/) to avoid impacting existing workload clusters. These new operators will not become active until existing workload clusters are upgraded or a new cluster is created.

## Management cluster-only components

There are a number of components that are not a part of a workload cluster release, such as our monitoring stack, or web UI. These components are deployed exclusively to the management cluster, and do not affect workload clusters.

> These components are deployed via app collections, which are a set of App Custom Resources, which are ultimately deployed via the App Platform. This collection contains the latest tagged version of each component, and is managed by our deployment component [draughtsman](https://github.com/giantswarm/draughtsman/).

Reviewer 2: ```suggestion
These components are deployed via app collections, which are a set of App custom resources, which are ultimately deployed via the App Platform. This collection contains the latest tagged version of each component, and is managed by our deployment component [draughtsman](https://github.com/giantswarm/draughtsman/).
```

These management cluster-only components are tested on our internal management clusters before releases are cut.

We plan on replacing draughtsman in Q2 2021, with a new component that more natively synchronises app collections to a management cluster, mostly to improve the scalability of the system.

## Configuration changes

Periodically, we need to update the configuration for management clusters.

> This configuration is currently held in a ConfigMap and a Secret stored on the Management Cluster. When updates are required, these resources are updated manually using our current tooling, from our central configuration repository.

Reviewer 2: ```suggestion
This configuration is currently held in a ConfigMap and a Secret stored on the management cluster. When updates are required, these resources are updated manually using our current tooling, from our central configuration repository.
```

These configuration files are watched by the App Platform, and any necessary applications are reconfigured on changes.

Changes to configuration are tested on our internal management clusters before being applied.

This configuration system is being updated in Q1 2021, with a system based on the new [config-controller](https://github.com/giantswarm/config-controller/). This will remove the need for manual updates of configuration, remove the shared ConfigMap and Secret, and allow for versioning of configuration. Overall, aiming to further increase the safety, reliability, and scalability of configuration changes.

## Infrastructure upgrades

The management cluster infrastructure, such as the Kubernetes cluster itself, or other supporting resources such as bastion hosts, are managed using Terraform.

> Updates to our Terraform playbooks are rolled out by CI as needed, with changes to our Terraform playbooks being tested on internal management clusters before being deployed.

Reviewer: I thought Terraform was modules and Ansible was playbooks? It's a good name but just not used to seeing it with TF.

Author: yeah you're fucking right, i'm confusing the two

> In 2021, we plan to start reusing our own workload clusters as the clusters that base management clusters, to both avoid having to maintain two methods of provisioning clusters, but also to gain the reliability benefits of our workload clusters.

Reviewer: The bit about operators being more dynamic was taken from https://www.giantswarm.io/management-cluster

@teemow asked if we could consolidate the docs with the website page.
https://github.com/giantswarm/giantswarm/issues/15700#issuecomment-780029135

Reviewer: Fine to drop it from my POV if you think it isn't needed.

Author: i'll take this into an issue in biscuit, because we can be a bit more aligned here, sure

Author: (see https://github.com/orgs/giantswarm/projects/125#card-55271871)
